scripts/benchmark/bench.py

```diff
@@ -0,0 +1,250 @@
+COVERAGE = os.getenv('COVERAGE', config.COVERAGE)
+TOKEN = os.getenv('TOKEN', config.TOKEN)
+DISTANT_BENCH_OUTPUT = os.getenv('DISTANT_BENCH_OUTPUT', config.DISTANT_BENCH_OUTPUT)
+OUTPUT_DIR = os.getenv('OUTPUT_DIR', config.OUTPUT_DIR)
+
+
+def get_coverage_start_production_date():
+    r = requests.get("{}/coverage/{}".format(NAVITIA_API_URL, COVERAGE), headers={'Authorization': TOKEN})
+    j = r.json()
+    return datetime.datetime.strptime(j['regions'][0]['start_production_date'], '%Y%m%d')
+
+
+def get_request_datetime(start_prod_date, days_shift, seconds):
+    return start_prod_date + datetime.timedelta(days=days_shift, seconds=seconds)
+
+
+def _call_jormun(url):
+    import time
+
+    time1 = time.time()
+    r = requests.get(url, headers={'Authorization': TOKEN})
+    time2 = time.time()
+    elapsed_time = (time2 - time1) * 1000.0
+    return r, elapsed_time
+
+
+def plot_per_request(data):
+    line_chart = pygal.Line(show_x_labels=False)
+    line_chart.title = 'Jormun Bench (in ms)'
+    for label, array in data:
+        line_chart.add(label, array)
+        line_chart.x_labels = map(str, range(0, len(array)))
+    line_chart.render_to_file('output_per_request.svg')
+
+
+def plot_normalized_box(data):
+    box = pygal.Box()
+    box.title = 'Jormun Bench Box'
+
+    for label, array in data:
+        box.add(label, array)
+
+    box.render_to_file('output_box.svg')
+
+
+def call_jormun(server, coverage, path, parameters, scenario_name, extra_args):
+    import os.path
+
+    # remove excessive /
+    normpath = os.path.normpath('/'.join(['coverage', coverage, path]))
+
+    url = "{}/{}?{}&_override_scenario={}&{}".format(server, normpath, parameters, scenario_name, extra_args)
+    ret, t = _call_jormun(url)
+    return url, t, ret.status_code
+
+
+def call_jormun_scenario(i, server, coverage, path, parameters, scenario, extra_args):
+    return i, call_jormun(server, coverage, path, parameters, scenario, extra_args)
+
+
+class Scenario(object):
+    def __init__(self, name, output_dir):
+        self.times = sortedcontainers.SortedDict()
+        if not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+
+        self.output = os.path.join(output_dir, "{}.csv".format(name))
+        with open(self.output, 'w') as f:
+            print("No,url,elapsed time,status_code", file=f)
+
+
+def submit_work(pool, reqs, extra_args, scenario):
+    for i, (path, params) in enumerate(reqs):
+        args = [i, NAVITIA_API_URL, COVERAGE, path, params, extra_args, scenario]
+        yield pool.submit(call_jormun_scenario, *args)
+
+
+def make_requests(input_file_name):
+    with fileinput.input(input_file_name) as f:
+        return [(req['path'], req['parameters']) for req in csv.DictReader(f)]
+
+
+def bench(args):
+    extra_args = args['--extra-args'] or ''
+    concurrent = int(args['--concurrent'] or 1)
+
+    reqs = make_requests(args['--input'] or '-')
```

Comment:
or '-'? If There's no input files with requests in it, there will be a string passed to `submit_work` ?

Reply:
that's the interesting thing of `fileinput`...   when the filename is '-', `Fileinput.input()` will read from `stdin`. yeah, that's not very trivial, i'll change it and add some comments